simplq/src/components/pages/Status/LeaveQueue.jsx

```diff
@@ -2,22 +2,23 @@ import React from 'react';
 import { useDispatch } from 'react-redux';
 import CloseIcon from '@material-ui/icons/Close';
 import { useHistory } from 'react-router-dom';
+import { useGetToken } from 'store/asyncActions';
 import SidePanelItem from '../../common/SidePanel/SidePanelItem';
 import { setInfoPopupMessage } from '../../../store/appSlice';
 
 export default (props) => {
+  const deleteToken = useGetToken(); // fix
   const history = useHistory();
   const dispatch = useDispatch();
-  const handleClick = () => {
-    props.leaveQueueHandler().then(() => {
-      dispatch(setInfoPopupMessage('Successfully left queue'));
-      history.push(`/`);
-    });
+  const onDeleteClick = () => {
+    dispatch(deleteToken(props.tokenId));
+    dispatch(setInfoPopupMessage('Successfully left queue'));
+    history.push(`/`);
```

Comment:
This is not fully correct. We should wait till the token is deleted and go back. I could add a comment and handle it later, but can't thing of a very elegant solution.

Reply:
You can dispatch any action from async action. Just watch for circular dependencies.

```js
//  deleteToken.js
      try {
        const response = await authedRequest;
        dispatch(setInfoPopupMessage('Successfully left queue'));
        history.goBack();
        return response;
      } catch (error) {
        return rejectWithValue({ message: `Some message...` });
      }
```